instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt

```diff
@@ -387,10 +393,127 @@ class RouteRefreshTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.ja
                 .first()
                 .navigationRoutes
 
+            // annotations
             assertEquals(201.673, requestedRoutes[0].getSumOfDurationAnnotationsFromLeg(1), 0.0001)
             assertEquals(189.086, refreshedRoutes[0].getSumOfDurationAnnotationsFromLeg(1), 0.0001)
+
+            // incidents
+            assertEquals(
+                listOf(listOf("9457146989091490", 400, 405)),
+                requestedRoutes[0].directionsRoute.legs()!![1].incidents()!!
+                    .extract({ id() }, { geometryIndexStart() }, { geometryIndexEnd() })
+            )
+            assertEquals(
+                listOf(listOf("9457146989091490", 396, 401)),
```

Comment:
do I get right that puck moved from 0 to 5 index (6 moves), but incident offset is changed from 400 to 396 (5 moves). Could you clarify why?

Reply:
No. That's exactly the data from refresh response. It's the next leg so it doesn't have to be changed. I changed the numbers in the refresh response just to emphasize that it's from the refresh, not the original route.